lib/framework/procbot.ts

```diff
@@ -22,10 +23,24 @@ import * as _ from 'lodash';
 import { ServiceConstructor, ServiceEmitRequest, ServiceEmitter, ServiceFactory,
 	ServiceListener, ServiceType } from '../services/service-types';
 import { Logger } from '../utils/logger';
-import { ConfigurationLocation, ProcBotConfiguration } from './procbot-types';
+import { ConfigurationLocation, ProcBotConfiguration, ProcBotErrorCode } from './procbot-types';
+
 const fsReadFile = Promise.promisify(FS.readFile);
 const exec: (command: string, options?: any) => Promise<{}> = Promise.promisify(ChildProcess.exec);
 
+export class ProcBotError extends TypedError {
+	private errorCode: ProcBotErrorCode;
+
+	constructor(code: ProcBotErrorCode, message: string) {
+		super(message);
+		this.errorCode = code;
+	}
+
+	public get code() {
```

Comment:
Error properties are public, and you shouldn't need to call a method to get any of them. ISTR this actually breaks ECMA error convention.